tests/integration-tests/tests/configure/test_pcluster_configure.py

```diff
@@ -161,3 +191,73 @@ def assert_config_contains_expected_values(
             validator.get("parameter_name")
         )
         assert_that(observed_value).is_equal_to(validator.get("expected_value"))
+
+
+def orchestrate_pcluster_configure_stages(
+    region,
+    key_name,
+    scheduler,
+    os,
+    instance,
+    vpc_id,
+    headnode_subnet_id,
+    compute_subnet_id,
+    vpc_stack,
+    omitted_subnets_num=0,
+):
+    compute_units = "vcpus" if scheduler == "awsbatch" else "instances"
+    # Due to the order of creation,PublicSubnet is the first in the list of subnets. Therefore it is the default value.
+    vpc_stack_public_subnet_id = vpc_stack.cfn_outputs["PublicSubnetId"]
+    # When there are omitted subnets, a note should be printed
+    omitted_note = "Note:  {0} subnet.+not listed.+".format(omitted_subnets_num) if omitted_subnets_num else ""
+    stage_list = [
+        {"prompt": r"AWS Region ID \[.*\]: ", "response": region},
+        {"prompt": r"EC2 Key Pair Name \[.*\]: ", "response": key_name},
+        {"prompt": r"Scheduler \[slurm\]: ", "response": scheduler},
+        {"prompt": r"Operating System \[alinux2\]: ", "response": os, "skip_for_batch": True},
+        {"prompt": fr"Minimum cluster size \({compute_units}\) \[0\]: ", "response": "1"},
+        {"prompt": fr"Maximum cluster size \({compute_units}\) \[10\]: ", "response": ""},
+        {"prompt": r"Master instance type \[t2\.micro\]: ", "response": instance},
+        {"prompt": r"Compute instance type \[t2\.micro\]: ", "response": instance, "skip_for_batch": True},
+        {"prompt": r"Automate VPC creation\? \(y/n\) \[n\]: ", "response": "n"},
+        {"prompt": r"VPC ID \[vpc-.+\]: ", "response": vpc_id},
+        {"prompt": r"Automate Subnet creation\? \(y/n\) \[y\]: ", "response": "n"},
+        {
+            "prompt": fr"{omitted_note}Master Subnet ID \[{vpc_stack_public_subnet_id}\]: ",
+            "response": headnode_subnet_id,
+        },
+        {
+            "prompt": fr"{omitted_note}Compute Subnet ID \[{vpc_stack_public_subnet_id}\]: ",
+            "response": compute_subnet_id,
+        },
+    ]
+    # When a user selects Batch as the scheduler, pcluster configure does not prompt for OS or compute instance type.
+    return [stage for stage in stage_list if scheduler != "awsbatch" or stage.get("skip_for_batch") is None]
+
+
+@pytest.fixture()
+def subnet_in_use1_az3(vpc_stack):
```

Comment:
Minor: hard coded implementation is ok for now, but we should try to make the function name and description more generic and add some To-do items.
So that if we choose to make this test generic or run it in another region in the future we can easily refactor the code.
For example we could do something like:
```
@pytest.fixture()
def subnet_without_instance_type(vpc_stack):
    # Subnet used to test the functionality of avoiding invalid subnet in AZs that do not have user specified instance types
    # Hard coding implementation to run on us-east-1 and create subnet in AZ without c5.xlarge or m6g.xlarge
    # Use use1_az3 as AZ, which does not have c5.xlarge or m6g.xlarge
    # To-do: extend code to support arbitrary region and arbitrary instance types
```